gcc/rust/rust-session-manager.cc

```diff
@@ -346,7 +401,13 @@ Session::handle_option (
     case OPT_frust_crate_:
       // set the crate name
       if (arg != nullptr)
-	ret = options.set_crate_name (arg);
+	{
+	  Error error ((Location ()), std::string ());
+	  if ((ret = validate_crate_name (arg, error)))
+	    options.set_crate_name (arg);
+	  else
+	    error.emit_error ();
```

Comment:
```suggestion
	  else {
	    rust_assert (!error.message.empty ());
	    error.emit_error ();
	  }
```
To make sure we filled the reference properly